### PR TITLE
Test the RUNTIME_IGNOREDOTFILES environment variable

### PR DESCRIPTION
**Pull request by mdimjasevic**
This PR tests the `RUNTIME_IGNOREDOTFILES` environment variable per issue #26.

### CODE REVIEW

File: test/Fencer/Rules/Test.hs [hunk 1]
```diff
@@ -26,20 +27,58 @@ tests = testGroup "Rule tests"
...
+test_rulesLoadRulesDotDirectory :: TestTree
+test_rulesLoadRulesDotDirectory =
+  testCase "Rules are loaded from a dot-directory" $
+    loadRules ".fencer-config" True
+
+
+-- | test that 'loadRulesFromDirectory' respects the
+-- RUNTIME_IGNOREDOTFILES environment variable.
+test_rulesLoadRulesRUNTIME_IGNOREDOTFILES :: TestTree
+test_rulesLoadRulesRUNTIME_IGNOREDOTFILES =
```
Comment (neongreen) [inline test/Fencer/Rules/Test.hs]: This test essentially checks that `getSettingsFromEnvironment` works. This isn't much.

I think that we should just have two tests for `loadRulesFromDirectory ... (#ignoreDotFiles True)`:

```
-- test that dot-directories are not ignored
<tmpdir>
    .dir1
        domain1.yml
```

```
-- test that dot-files are ignored
<tmpdir>
    dir2
        .domain2.yml
```

If we want to check that `getSettingsFromEnvironment` uses right variable names, we can do it separately (though I think we don't have to do this at all).

Comment (mdimjasevic) [reply test/Fencer/Rules/Test.hs]: Addressed! I got rid of `getSettingsFromEnvironment` in the test.

File: test/Fencer/Rules/Test.hs [hunk 2]
```diff
@@ -26,20 +28,70 @@ tests = testGroup "Rule tests"
...
+  testCase "Rules are loaded from a dot-directory" $
+    expectLoadRules
+      (#ignoreDotFiles True)
+      (#files
+        [ (".domain1/config1.yml", domain1Text)
+        , (".domain2/config2.yaml", domain2Text) ]
+      )
+      (#result [domain1, domain2])
+
+-- | test that 'loadRulesFromDirectory' respects the
```
Comment (neongreen) [inline test/Fencer/Rules/Test.hs]: "respects ignoreDotFiles"?

Comment (neongreen) [reply test/Fencer/Rules/Test.hs]: Btw there should also be a test for `#ignoreDotFiles False`.

Comment (mdimjasevic) [reply test/Fencer/Rules/Test.hs]: Addressed in the upcoming commit.

**Comment by neongreen** [comment]
This includes #77, right? I think in this case you could just add the test to #77 directly.

File: test/Fencer/Rules/Test.hs [hunk 3]
```diff
@@ -26,20 +28,85 @@ tests = testGroup "Rule tests"
...
+      (#ignoreDotFiles True)
+      (#files
+        [ (".domain1" </> "config1.yml", domain1Text)
+        , ("domain2" </> "config2.yaml", domain2Text) ]
+      )
+      (#result [domain2])
+
+-- | test that 'loadRulesFromDirectory' correctly implements the case
+-- RUNTIME_IGNOREDOTFILES=true.
+test_rulesLoadRulesRUNTIME_IGNOREDOTFILEStrue :: TestTree
```
Comment (neongreen) [inline test/Fencer/Rules/Test.hs]: Can you rename it to `test_rulesLoadRules_ignoreDotFiles` or something? It is not related to the env var anymore, it's a unit test for `loadRulesFromDirectory` basically.

**Review by neongreen — APPROVED** [review]

File: test/Fencer/Rules/Test.hs (hunk 3 above)
Comment (mdimjasevic) [reply test/Fencer/Rules/Test.hs]: Sure! Done in an upcoming commit.

**Comment by mdimjasevic** [comment]
> This includes #77, right? I think in this case you could just add the test to #77 directly.

Yes, it pulls from the same branch that PR #77 is based on, but I kept the two PRs in separate branches. They started out separately, but while switching back and forth I realized I need stuff from #77 in here as well. To avoid merging back to the original branch, I'll just keep it the way it is and merge it to master as soon as CI passes.

File: README.md [hunk 4]
```diff
@@ -137,7 +137,7 @@ Fencer-specific environment variables are:
 - `RUNTIME_IGNOREDOTFILES` - A flag indicating whether to ignore files
   with names starting with a dot (hidden files on Linux-based systems
   and macOS). It can be `True` or `False`. The default value is
-  `False`.
+  `False`. Directories with names starting in a dot are not ignored.
```
Comment (neongreen) [inline README.md]: They are.

So, this line should be removed and the description should say "A flag indicating whether to ignore files and directories"

Comment (mdimjasevic) [reply README.md]: Fixed. Thank you for spotting this!

**Pull request merged**